lib/aws_cloudwatch.dart

```diff
@@ -640,8 +640,7 @@ class CloudWatch {
   bool _checkError(dynamic error) {
     if (error != null) {
       if (!raiseFailedLookups &&
-          error is SocketException &&
-          error.message.startsWith('Failed host lookup')) {
+          error.message.contains('XMLHttpRequest error')) {

```

Comment:
What conditions caused the `XMLHttpRequest error`? 

The `SocketException Failed host lookup` is there to catch when the user doesnt have internet and to stop it from throwing a fit. I think we still need that but if there is another similar case Id love to add it or make another if statement for it.

Reply:
so, I tried to possibly recreate a scenario wherein I tried to make a request with no network, added a debugger there and the resultant error message was ```XMLHttpRequest error.``` exactly.

So I put this as a condition there.

Reply:
I cant reproduce that `XMLHttpRequest error`. What browser were you using? I assume this is still on your mac?  

As a side note, `error` is dynamic so the compiler didnt complain but if we dont verify the type we cant assume it will have a property `message`. It would be better to call `error.toString().contains('XMLHttpRequest error')` which will never fail.

Reply:
Yes, this was on mac.

Sure, I will update the condition to this.

Reply:
Thanks for the info! I was able to reproduce it when I disconnected my internet instead of disabling the adapter. The `XMLHttpRequest error` does indeed occur when there is no internet but only on web. On mobile we still need the `Failed host lookup`. 

Can you OR `error.toString().contains('Failed host lookup')` with the new check you added? After that I think we're all good 😁

Reply:
Great, thanks for highlighting this, I have appended the condition to the existing one, please have a look into the changes.